src/include/processor/operator/recursive_extend/path_scanner.h

```diff
@@ -27,16 +27,24 @@ struct BaseFrontierScanner {
     virtual ~BaseFrontierScanner() = default;
 
     size_t scan(common::table_id_t tableID, common::ValueVector* pathVector,
-        common::ValueVector* dstNodeIDVector, common::sel_t& offsetVectorPos,
-        common::sel_t& dataVectorPos);
+        common::ValueVector* dstNodeIDVector, common::ValueVector* pathLengthVector,
+        common::sel_t& offsetVectorPos, common::sel_t& dataVectorPos);
 
     void resetState(const BaseBFSMorsel& bfsMorsel);
 
-private:
+protected:
     virtual void initScanFromDstOffset() = 0;
     virtual void scanFromDstOffset(common::table_id_t tableID, common::ValueVector* pathVector,
-        common::ValueVector* dstNodeIDVector, common::sel_t& offsetVectorPos,
-        common::sel_t& dataVectorPos) = 0;
+        common::ValueVector* dstNodeIDVector, common::ValueVector* pathLengthVector,
+        common::sel_t& offsetVectorPos, common::sel_t& dataVectorPos) = 0;
+
+    inline void writeDstNodeOffsetAndLength(common::ValueVector* dstNodeIDVector,
+        common::ValueVector* pathLengthVector, common::sel_t& offsetVectorPos,
+        common::table_id_t tableID) {
+        dstNodeIDVector->setValue<common::nodeID_t>(
+            offsetVectorPos, common::nodeID_t{currentDstOffset, tableID});
+        pathLengthVector->setValue<int64_t>(offsetVectorPos, (int64_t)k);
+    }
 };
 
 /*
```

Comment:
In this comment should you say `DstNodeScanner scans dst node offsets and lengths of paths only."?